00_ES6/00_start/src/pages/chat/chat.container.jsx

```diff
@@ -1,8 +1,26 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import { Link } from 'react-router-dom';
+import { SessionContext } from '../../common';
 
-export const ChatContainer = () =>
+const ChatContainerInner = (props) =>
   <React.Fragment>
     <h1>Hello from chat page</h1>
+    <p>nickname: {props.nickname}</p>
+    <p>room: {props.room}</p>
     <Link to="/">Navigate back to lobby</Link>
-  </React.Fragment>
\ No newline at end of file
+  </React.Fragment>
+
+ChatContainerInner.propTypes = {
+  nickname : PropTypes.string.isRequired,
+  room: PropTypes.string.isRequired,
+};
+
+
+export const ChatContainer = () =>
```

Comment:
We could create an HOC to feed SessionContext.Consumer if we are going to use in several components:

### ./src/common/sessionConsumer.js
```javascript
import { SessionContext } from './context';

const SessionConsumer = SessionContext.Consumer;

export const withSessionContext = (Component) => (props) => (
   <SessionConsumer>
      {
         ({ nickname, room, setChatSessionInfo }) => (
            <Component
                  {...props}
                  nickname={nickname}
                  room={room}
                  setChatSessionInfo={setChatSessionInfo}
            />
         )
      }
  </SessionConsumer>
);
```

Reply:
That's a good point, mmm, if we go that way:

- Maybe it should be a good idea to group _nickname, rom, setChatSessionInfo_ in an object something like sessionInfo (and inside you get the three props).

- Best place to store this I think should be the common file where we defined SessionInfoProvider

Then once we port to redux, we will just get rid of the HOC and move everything to redux, I think it will be easier for the student to remove just the Hoc call than the whole thing, I think is a good idea.

What do you think?

Reply:
Yes for sure:

- One property sessionInfo to get less code

- Sure, near from provider and context.

- Exactly, refactor is easier.

Reply:
I prefer HOC, by the way, I do not feel confortable with Session on naming but I can not figure out something better.

Reply:
Updated I think with your update now is looking great, really good idea @Nasdan

Reply:
Hoc helps a lot once implemented looks cleanner.
@JaimeSalas about naming agree, not sure abot a better name, but maybe for a small-medium app, we could start create a session context object (it would be like generic session info), for80% of the apps that would be fine (usually we need to store, user name, last search small stuff), if it starts to grow I would cosider rather than split into more context, evaluate to introduce redux (I think is justified, I see context for small stuff, not complex structures)